opencti-platform/opencti-graphql/src/utils/markingDefinition-utils.js

```diff
@@ -0,0 +1,86 @@
+import * as R from 'ramda';
+import { getEntitiesMapFromCache } from '../database/cache';
+import { ENTITY_TYPE_MARKING_DEFINITION } from '../schema/stixMetaObject';
+import { SYSTEM_USER } from './access';
+import { UPDATE_OPERATION_ADD, UPDATE_OPERATION_REMOVE, UPDATE_OPERATION_REPLACE } from '../database/utils';
+
+export const cleanMarkings = async (context, values) => {
+  const markingsMap = await getEntitiesMapFromCache(context, SYSTEM_USER, ENTITY_TYPE_MARKING_DEFINITION);
+  const defaultMarkingValues = values?.map((d) => markingsMap.get(d) ?? d) ?? [];
```

Comment:
I think, it's better to return always the same type and filter on undefined
```suggestion
  const defaultMarkingValues = values?.map((d) => markingsMap.get(d)).filter((v) => v !== undefined) ?? [];
```

Reply:
After our discussion, we keep the current code :)